### DISCUSSION TITLE
Quiz測驗第2615答案

### DISCUSSION THREAD

**Discussion by B1nn9** [1101-GS4719-C/Python-Community-GS4719-C#1673] [General]
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/144880346/305ca6e2-9b4f-42bf-9c0d-6f5d1c9df3d1)
C、D應該都是正確的選項吧？

**Comment by lazywulf** [comment]
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/103130674/a58ed1d7-16a5-4662-b7d6-e4bab6cdea01)
不行做 slicing 啊。集合欸。切片啥意思？

**Reply by lazywulf** [reply]
對耶。他說的是「不能」。抱歉喔。

**Comment by 1chooo** [comment]
同樣回覆在 #1674 

我在想切片操作的意思是不是像是 `list` 的 `test_list[1, 3]` 的操作，如果是的話，那 `set` 確實沒有切片的操作，如以下的範例程式碼：

```py
test_set = {1, 2, 3, 4, 5}

print(test_set)
print(test_set[1, 3])
```

可以看出第二行能印出來，但是在第三行執行會出錯。

```shell
{1, 2, 3, 4, 5}
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<my_apth>/test.py](https://file+.vscode-resource.vscode-cdn.net/<my_apth>/test.py) in line 3
      [12](file:///<my_apth>/test.py?line=11) test_set = {1, 2, 3, 4, 5}
      [13](file:///<my_apth>/test.py?line=12) print(test_set)
----> [14](file:///<my_apth>/test.py?line=13) print(test_set[1, 3])

TypeError: 'set' object is not subscriptable
```

### 後續
不過我有去查 stackoverflow 其實 `set` 也可以用 slice 的操作。
- 原文連結：[What is a pythonic way of slicing a set?](https://stackoverflow.com/questions/40736681/what-is-a-pythonic-way-of-slicing-a-set)

原文中的解法參考：
```py
import itertools

some_data = [1, 2, 4, 1, 6, 23, 3, 56, 6, 2, 3, 5, 6, 32, 2, 12, 5, 3, 2]
big_set = set(some_data)
small_set = set(itertools.islice(big_set, 5))

print(small_set)    # {32, 1, 2, 3, 4}
```

所以其實 python 中的 `set` 應該還是可以有 slice 操作，不過可能需要搭配其他套件使用就是了！

（大家如果有想法都非常歡迎進一步討論！）
